File: C/bmi.c
#include <stdio.h>

int main() {
    int age;
    char gender;
    float weight, height, height_cm, bmi;

    // Taking input
    printf("Enter your age: ");
    scanf("%d", &age);
    
    printf("Enter your gender (M/F/O): ");
    scanf(" %c", &gender); // Space before %c to handle newline character

    printf("Enter your weight in kg: ");
    scanf("%f", &weight);
    
    printf("Enter your height in cm: ");
    scanf("%f", &height);

    // Convert height from cm to meters
    height_cm = height / 100;

    // Calculate BMI
    bmi = weight / (height_cm * height_cm);

    // Display BMI result
    printf("Your BMI is: %.2f\n", bmi);

    // Provide suggestions
    if (bmi < 18.5)
        printf("You are underweight. Try to eat a balanced diet.\n");
    else if (bmi >= 18.5 && bmi < 24.9)
        printf("You have a normal weight. Keep maintaining your lifestyle!\n");
    else if (bmi >= 25 && bmi < 29.9)
        printf("You are overweight. Consider regular exercise and a healthy diet.\n");
    else
        printf("You are obese. Seek medical advice for better weight management.\n");

    return 0;
}

File: dart/bmi.dart
import 'dart:io';

void main() {
  
  stdout.write("Enter your age: ");
  int age = int.parse(stdin.readLineSync()!);

  stdout.write("Enter your gender (M/F/O): ");
  String gender = stdin.readLineSync()!.toUpperCase();

  stdout.write("Enter your weight (kg): ");
  double weight = double.parse(stdin.readLineSync()!);

  stdout.write("Enter your height (feet): ");
  double heightFeet = double.parse(stdin.readLineSync()!);

  
  double heightMeters = heightFeet * 0.3048;


  double bmi = weight / (heightMeters * heightMeters);
  bmi = double.parse(bmi.toStringAsFixed(2)); 

  String category = "";
  String suggestion = "";

  if (bmi < 18.5)
  {
    category = "Underweight";
    suggestion = "You should consider increasing your calorie intake and focusing on a balanced diet.";
  } 
  else if (bmi >= 18.5 && bmi < 24.9)
  {
    category = "Normal weight";
    suggestion = "Great job! Maintain a healthy lifestyle and regular exercise.";
  }
   else if (bmi >= 25 && bmi < 29.9)
  {
    category = "Overweight";
    suggestion = "Consider a healthy diet and regular physical activity to manage your weight.";
  } 
  else
   {
    category = "Obese";
    suggestion = "It's important to consult a doctor or dietitian for a healthy weight management plan.";
  }


  print("Age: $age years");
  print("Gender: $gender");
  print("Weight: $weight kg");
  print("Height: $heightFeet feet ($heightMeters meters)");
  print("Your BMI: $bmi");
  print("Category: $category");
  print("Suggestion: $suggestion");
}
